commands/add.ts

```diff
@@ -80,16 +83,20 @@ export default class Add extends BaseCommand {
   /**
    * Install the package using the selected package manager
    */
-  async #installPackage(npmPackageName: string) {
+  async #installPackage(npmPackageName: string, npmPackageVersion: string) {
     const colors = this.colors
     const spinner = this.logger
       .await(`installing ${colors.green(this.name)} using ${colors.grey(this.packageManager!)}`)
       .start()
 
     spinner.start()
 
+    const packageToInstall = npmPackageVersion
+      ? `${npmPackageName}@${npmPackageVersion}`
+      : npmPackageName
+
     try {
-      await installPackage(npmPackageName, {
+      await installPackage(packageToInstall, {
```

Comment:
This doesn't work for the packages that are filesystem based installs, like those used in the tests. Really the actual installation should probably be mocked out in most of the tests, since we can probably assume that `installPackage` does its thing.

Reply:
The reason we don't mock is, because we have been bitten in the past with the `install-pkg` package. They swapped the underlying fetch library that had different timeout settings and suddenly all installations taking longer than x timeout started failing.